### ISSUE TITLE
Make component and import into another SPFx  project

### ISSUE THREAD

**Issue by necik11** [SharePoint/sp-dev-docs#8869]
### What type of issue is this?

Question

### What SharePoint development model, framework, SDK or API is this about?

💥 SharePoint Framework

### Target SharePoint environment

SharePoint Online

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [X] 💥 Microsoft Edge
- [X] 💥 Google Chrome
- [ ] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [ ] not applicable
- [ ] other (enter in the "Additional environment details" area below)

### Additional environment details

- browser version
  - Chrome 111.0.5563.146
- SPFx version
  - 1.16.1  
- Node.js version
  - v16.14.1

### Issue description

I apologize in advance as this may be a trivial question, but I would like to advise on best practices if I would like to:

The creation of a **component** (I'm not saying that I have to write it in React) that I would like to have be reusable in SPFx and, for example, in another project ... even after build in html itself.

I tried this procedure:
[E.g. complete simple project creation](https://create-react-app.dev/docs/getting-started)

`npx create-react-app my-app
`
[Then create production build:](https://create-react-app.dev/docs/deployment)
`npm run build`

and et end just import js and css files into tsx file:

`
require('./static/css/main.073c9b0a.css');
require('./static/js/787.5c78d9aa.chunk.js');
require('./static/js/main.1d04d201.js');
`
And if it is necessary that these 2 react applications have the same version of all libraries + versions of react 17 -18.

I got error even i demote react version...
I tried several procedures, but each time there was an error.. now i got:
`Could not load webpartName in require. Error: Minified React error #200; `

Specifically, I would like to connect
[pdf-lib](https://pdf-lib.js.org/docs/api/classes/pdfsignature)
[signature_pad](https://github.com/szimek/signature_pad)
and HTTP queries (propably fetch or some library).

It's all pure JS and CSS, so how do you make a jsx file usable everywhere? Especially in SPFx! Thanks for answers 👍

**Comment by andrewconnell** [comment]
Yup, suggest you take a look at the project https://pnp.github.io/sp-dev-fx-controls-react/ that does just this.

> And if it is necessary that these 2 react applications have the same version of all libraries + versions of react 17 -18.

You shouldn't use a version of React that's not in line with the version of SPFx you're using. React is already on the page, added by SPFx. You can't impact that. If you try to add another version of React to the page, you'll have two instances of React on the page and that's not recommended (and can cause all sorts of issues).

**Comment by AJIXuMuK** [comment]
@necik11 - you can also check documentation on how do add [externals](https://learn.microsoft.com/en-us/sharepoint/dev/spfx/web-parts/basics/add-an-external-library) to your app.

Please, let us know if it answers your questions.

**Comment by necik11** [comment]
That was the first thing what i have i did "check documentation".
And I tried to import js and css files from another react project that I published via `npm run build`.

Anyway, even when I edited `config.json` and added `require()` it didn't work...also returned an error

So i choose another solution:
1) I found the exact version of react in package.json.
2) Install latest react app: `npx create-react-app my-app`
3) [Downgrade version](https://dev.to/ifeanyichima/how-to-downgrade-from-react-18-to-1702-818)
4) Install sass `npm i sass --save-dev`
5) Run app `npm start`
6) When i build by: `npm run build`  i got build folder which i can distribute to another systems.... And for another spfx solution i just copy folder of component...
7) For future i will use publishing package

Thanks much for tips! 👌
